File: tags/milestone1/Noxa.Emulation.Psp.Bios.NativeHLE/KCallback.cpp
// ----------------------------------------------------------------------------
// PSP Player Emulation Suite
// Copyright (C) 2006 Ben Vanik (noxa)
// Licensed under the LGPL - see License.txt in the project root for details
// ----------------------------------------------------------------------------

#include "Stdafx.h"
#include "KernelHelpers.h"
#include "KCallback.h"
#include "KThread.h"
#include <malloc.h>
#include <string.h>

using namespace System;
using namespace Noxa::Emulation::Psp;
using namespace Noxa::Emulation::Psp::Bios;

KCallback::KCallback( char* name, KThread* thread, uint address, uint commonAddress )
{
	Name = _strdup( name );
	Thread = thread;
	Address = address;
	CommonAddress = commonAddress;
}

KCallback::~KCallback()
{
	SAFEFREE( Name );
}

File: trunk/Noxa.Emulation.Psp.Bios.ManagedHLE/Kernel Types/KCallback.cs
// ----------------------------------------------------------------------------
// PSP Player Emulation Suite
// Copyright (C) 2006 Ben Vanik (noxa)
// Licensed under the LGPL - see License.txt in the project root for details
// ----------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Text;

namespace Noxa.Emulation.Psp.Bios.ManagedHLE
{
	class KCallback
	{
		public Kernel Kernel;

		public string Name;
		public KThread Thread;
		
		//typedef int (*SceKernelCallbackFunction)(int count, int arg, void *common);

		public uint Address;
		public uint CommonAddress;

		public uint NotifyCount;
		public uint NotifyArguments;

		public KCallback( Kernel kernel, string name, KThread thread, uint address, uint commonAddress )
		{
			Kernel = kernel;
			Name = name;
			Thread = thread;
			Address = address;
			CommonAddress = commonAddress;

			NotifyCount = 0;
			NotifyArguments = 0;
		}
	}
}
